CenterForOpenScience/osf.io
# [IN-149][OSF] Add AbstractProvider Model

## Purpose

[WIP] Generalize PreprintProvider model into a class that can be used
for other upcoming types of providers (such as CollectionsProvider) 

## Changes

- Add migration to create AbstractProvider table
- Update tests
- Update modules previously using PreprintProvider  model

## QA Notes
- Mostly backend changes. 
- Regression tests for Preprints, PreprintProvider to make sure nothing
is broken by this change.
- Maybe API testing on preprints/preprint_providers routes

## Side Effects

## Ticket

[IN-149](https://openscience.atlassian.net/browse/IN-149)
